src/Task/Ecs.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Runner\TaskResultInterface;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Ecs task.
+ */
+class Ecs extends AbstractExternalTask
+{
+    public function getName(): string
+    {
+        return 'ecs';
+    }
+
+    public function getConfigurableOptions(): OptionsResolver
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'whitelist_patterns' => [],
+            'fix' => false,
+            'clear-cache' => false,
+            'no-progress-bar' => true,
+            'config' => null,
+            'level' => null,
+        ]);
+
+        $resolver->addAllowedTypes('whitelist_patterns', ['array']);
+        $resolver->addAllowedTypes('fix', ['bool']);
+        $resolver->addAllowedTypes('clear-cache', ['bool']);
+        $resolver->addAllowedTypes('no-progress-bar', ['bool']);
+        $resolver->addAllowedTypes('config', ['null', 'string']);
+        $resolver->addAllowedTypes('level', ['null', 'string']);
+
+        return $resolver;
+    }
+
+    public function canRunInContext(ContextInterface $context): bool
+    {
+        return $context instanceof GitPreCommitContext || $context instanceof RunContext;
+    }
+
+    public function run(ContextInterface $context): TaskResultInterface
+    {
+        $files = $context->getFiles()->name('*.php');
```

Comment:
Can you add the 'triggered by php' option?
See other tasks (like PhpCsFixerV2):
```
$files = $context->getFiles()->extensions($config['triggered_by']);
```

Don't forget to add it to the doc as well 😉

Reply:
Yes good idea :)